native/src/main/java/io/ballerina/stdlib/grpc/Message.java

```diff
@@ -177,6 +205,59 @@ public Message(
         } else if (type.getTag() == TypeTags.ARRAY_TAG) { // for array values inside structs
             bArray = ValueCreator.createArrayValue(TypeCreator.createArrayType(PredefinedTypes.TYPE_ANYDATA));
             bMessage = bArray;
+        } else if (isAnyTypedMessage && input != null) {
+            int typeUrlTag = input.readTag();
+
+            if (typeUrlTag == 10) {
```

Comment:
Shall we replace `10` with a constant here

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/509/commits/a449ffc6ce0923859a0c263b340d246bbc5fe8cf